### ISSUE TITLE
QA Process - non-separated tokens

### ISSUE THREAD

**Issue by NathanD38** [IAHLT/UD_Hebrew#4]
@amir-zeldes
We have some sentences in which the original source included a non-separated token that should have been separated.
For example:

`# sent = הוא אומר עודכי למערכת יכולת ייחודית לבחינת “טביעת האצבע” של החיישן ובכך לזהות אנומליות.
`

> 1	הוא	הוא	PRON	PRON	Gender=Masc|Number=Sing|Person=3|PronType=Prs	2	nsubj	_	_
2	אומר	אמר	VERB	VERB	Gender=Masc|HebBinyan=PAAL|Number=Sing|Person=3|Tense=Pres|VerbForm=Part|Voice=Act	0	root	_	_
**3	עודכי	עודכי	X	X	Typo=Yes	2	advmod	_	CorrectForm=עוד כי**
4	ל	ל	ADP	ADP	Definite=Def|PronType=Art	5	case	_	_
5	מערכת	מערכת	NOUN	NOUN	Gender=Fem|Number=Sing	2	obl	_	_
6	יכולת	יכולת	NOUN	NOUN	Gender=Fem|Number=Sing	5	nmod	_	_
7	ייחודית	ייחודי	ADJ	ADJ	Gender=Fem|Number=Sing	6	amod	_	_
8	ל	ל	ADP	ADP	_	9	case	_	_
9	בחינת	בחינה	NOUN	NOUN	Definite=Cons|Gender=Fem|Number=Sing	6	nmod	_	_
10	“	“	PUNCT	PUNCT	_	11	punct	_	_
11	טביעת	טביעה	NOUN	NOUN	Definite=Cons|Gender=Fem|Number=Sing	9	compound	_	_
12	ה	ה	DET	DET	Definite=Def|PronType=Art	13	det	_	_
13	אצבע	אצבע	NOUN	NOUN	Gender=Fem|Number=Sing	11	compound	_	_
14	”	”	PUNCT	PUNCT	_	11	punct	_	_
15	של	של	ADP	ADP	Case=Gen	17	case	_	_
16	ה	ה	DET	DET	Definite=Def|PronType=Art	17	det	_	_
17	חיישן	חיישן	NOUN	NOUN	Gender=Masc|Number=Sing	11	nmod	_	_
18	ו	ו	CCONJ	CCONJ	_	20	cc	_	_
19	ב	ב	ADP	ADP	_	20	case	_	_
20	כך	כך	PRON	PRON	PronType=Dem	21	obl	_	_
21	לזהות	זיהה	VERB	VERB	HebBinyan=PIEL|VerbForm=Inf|Voice=Act	9	acl	_	_
22	אנומליות	אנומליה	NOUN	NOUN	Gender=Fem|Number=Plur	21	obj	_	_
23	.	.	PUNCT	PUNCT	_	2	punct	_	_

The upos of עודכי is X with the feature CorrectForm=עוד כי
The correct form would separate the tokens into an ADV+advmod and SCONJ+mark.

Should we separate them to render the correct analysis possible, 
or should we leave them as corrupt, while keeping the original text intact?

A related question to the structure of this sentence is a possible eliptical **Yesh** between tokens 5 and 6.
If we treat this sentence as one with an elipsis, which token is the head of the ccomp? 
Do we have ccomp(amar,yecholet) or ccomp(amar, maarechet)?
If it is ccomp(amar,yecholet), what is the deprel between yecholet and maarechet?
If it is ccomp(amar, maarechet), then it is the obl argument of an implicit **Yesh**. with nsubj(maarechet, yecholet).

In case we don't treat sans-Yesh sentences as eliptical, what should be the deprels within the ccomp, and which head is its root?

Thanks!
Netanel

**Comment by amir-zeldes** [comment]
Yes, the tokens should be separated, and the first should have in MISC `CorrectSpaceAfter=Yes|SpaceAfter=No`, see https://universaldependencies.org/u/overview/typos.html

For yesh, if we treat it as elliptical (which might be easiest), the canonical analysis is to promote based on the hierarchy (so ccomp would attach to yecholet), and attach maarechet as `orphan` to yecholet. A non-elliptical analysis could view yecholet as the ccomp head, and attach maarechet as `obl` (prepositional dependent of a predicate noun), but maybe `orphan` is a better way of capturing that they are both arguments of the same unexpressed predication.

**Comment by NathanD38** [comment]
@amir-zeldes 
Thank you! I was not aware of those features, suitable for dealing with such cases.
The promotion of `nsubj` "yecholet" in such constructions is more clear to me now.

This is the file now, which I hope it correct.

> 1	הוא	הוא	PRON	PRON	Gender=Masc|Number=Sing|Person=3|PronType=Prs	2	nsubj	_	_
2	אומר	אמר	VERB	VERB	Gender=Masc|HebBinyan=PAAL|Number=Sing|Person=3|Tense=Pres|VerbForm=Part|Voice=Act	0	root	_	_
**3	עוד	עוד	ADV	ADV	_	2	advmod	_	CorrectSpaceAfter=Yes|SpaceAfter=No
4	כי	כי	SCONJ	SCONJ	_	7	mark	_	_**
5	ל	ל	ADP	ADP	Definite=Def|PronType=Art	6	case	_	_
6	מערכת	מערכת	NOUN	NOUN	Gender=Fem|Number=Sing	7	orphan	_	_
7	יכולת	יכולת	NOUN	NOUN	Gender=Fem|Number=Sing	2	ccomp	_	_
8	ייחודית	ייחודי	ADJ	ADJ	Gender=Fem|Number=Sing	7	amod	_	_
9	ל	ל	ADP	ADP	_	10	case	_	_
10	בחינת	בחינה	NOUN	NOUN	Definite=Cons|Gender=Fem|Number=Sing	7	**nmod**	_	_
11	“	“	PUNCT	PUNCT	_	12	punct	_	_
12	טביעת	טביעה	NOUN	NOUN	Definite=Cons|Gender=Fem|Number=Sing	10	compound	_	_
13	ה	ה	DET	DET	Definite=Def|PronType=Art	14	det	_	_
14	אצבע	אצבע	NOUN	NOUN	Gender=Fem|Number=Sing	12	compound	_	_
15	”	”	PUNCT	PUNCT	_	12	punct	_	_
16	של	של	ADP	ADP	Case=Gen	18	case	_	_
17	ה	ה	DET	DET	Definite=Def|PronType=Art	18	det	_	_
18	חיישן	חיישן	NOUN	NOUN	Gender=Masc|Number=Sing	12	nmod:poss	_	_
19	ו	ו	CCONJ	CCONJ	_	22	cc	_	_
20	ב	ב	ADP	ADP	_	21	case	_	_
21	כך	כך	PRON	PRON	PronType=Dem	22	obl	_	_
22	לזהות	זיהה	VERB	VERB	HebBinyan=PIEL|VerbForm=Inf|Voice=Act	7	**conj**	_	_
23	אנומליות	אנומליה	NOUN	NOUN	Gender=Fem|Number=Plur	22	obj	_	_
24	.	.	PUNCT	PUNCT	_	2	punct	_	_

I attach a picture of the tree and MISC features of 3. I do not know how to link to trees here yet, 
and it's not in any Arborator interface, but a part of the QA connlu file.
I just copied it to a temporary file which I changed back without saving.

![image](https://user-images.githubusercontent.com/84139569/132708332-99055a48-3b3a-4a53-8d00-8a084c5f9a1d.png)

![image](https://user-images.githubusercontent.com/84139569/132708718-093f306d-79e1-4024-b482-8badcea06fc9.png)

Two things I am not sure about:

1. Do we connect the infinitive "lezahot" in 22 with `conj` to "yecholet" or with `acl`? Or perhaps it is connected to "bxinat" in 10?
2. Does the NOUN complement in 10 receive `obl` from "yecholet" or `nmod`? 
This is relevant whether we have an explicit or implicit Yesh.

**Comment by amir-zeldes** [comment]
Yes, looks good. I think it's conj because "בכך" is an adjunct and could be dropped, and the tree would have to stay intact. I think 10 is nmod (יכולת לבחינת X is an NP, interrogable, movable etc.)

**Comment by NathanD38** [comment]
@amir-zeldes 
Thank you!
